obelix/idefix/services/github/issue.go

```diff
@@ -0,0 +1,101 @@
+package github
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/Shopify/sarama"
+	redisv8 "github.com/go-redis/redis/v8"
+	"idefix/operator"
+	"idefix/producer"
+	"idefix/redis"
+)
+
+var ErrNoIssues = fmt.Errorf("no issues in the repository")
+
+type Issues struct {
+	Issues []Issue `json:"issues,omitempty"`
+}
+
+type Issue struct {
+	ID            int         `json:"id"`
+	NodeID        string      `json:"node_id"`
+	URL           string      `json:"url"`
+	RepositoryURL string      `json:"repository_url"`
+	PullRequest   interface{} `json:"pull_request,omitempty"`
+}
+
+func (i *Issues) Parse(data []byte) error {
+	err := json.Unmarshal(data, &i.Issues)
+	if err != nil {
+		return fmt.Errorf("failed to Unmarshal %w", err)
+	}
+
+	return nil
+}
+
+func (i *Issues) findLatestIssue() (*Issue, bool) {
+	for _, elem := range i.Issues {
+		if elem.PullRequest == nil {
+			return &elem, true
+		}
+	}
+
+	return nil, false
+}
+
+func (i *Issues) GetRedisState(rc *redis.Client, key string) (string, error) {
+	state, err := rc.GetKey(key)
+	if errors.Is(err, redisv8.Nil) {
+		elem, ok := i.findLatestIssue()
+		if !ok {
+			return "", ErrNoIssues
+		}
+
+		err = rc.SetKey(key, elem.URL)
+		if err != nil {
+			return "", fmt.Errorf("failed to set value in redis: %w", err)
+		}
```

Comment:
Since you are doing `rc.SetKey(key, elem.URL)`, you reset err to nil if there is no error so you will not fall in the 2nd if statement.
Indeed, if `error.Is(err, redisv8.Nil)` is false, you'll go to the if statement.
So again, the else is useless

Reply:
3e3807427ed531fc112e7b992b1056cb0f21372f